src/go/k8s/pkg/resources/configmap.go

```diff
@@ -149,7 +178,16 @@ func (r *ConfigMapResource) createConfiguration() *config.Config {
 		})
 	}
 
-	return cfgRpk
+	return cfgRpk, nil
+}
+
+func (r *ConfigMapResource) getSecretValue(ctx context.Context, nsName types.NamespacedName, key string) (string, error) {
+	var secret corev1.Secret
+	err := r.Get(ctx, nsName, &secret)
+	if err != nil {
+		return "", err
+	}
+	return string(secret.Data[key]), nil
```

Comment:
what if the data does not contain the key?

Reply:
It would return an empty string, which we'd then give to Redpanda.

Reply:
I added a check for empty string. It returns an error now.